### ISSUE TITLE
Feature Request: Implement InlineSpan for TextFields

### ISSUE THREAD

**Issue by steeling** [flutter/flutter#82595]
I'm using a TextSpan where I alternate between static text and user input (think of the game madlibs). The issue is that the TextField, when wrapped in a WidgetSpan, is treated as a single block, whereas a TextSpan, can be line broken on spaces.

So I may have text that ends up like the following (a '-' represents static text and a '+' represents text from a textfield). and the '|' represents the contraints of the textspan:
```
---    ---  --- |
-- --  ---      | 
++ ++ + ++ + ---|
--  ---  ----   |
```
As you can see, some of the +'s would clearly fit on the previous line, but since  a widgetspan is treated as one line it can't. I'd be  happy to accomplish the same goal in some other mechanisms than TextSpan by the way.

I've thought about using a single textfield, and a custom inputformatter to prevent edits within the static portions, but it means that the cursor can go over the static text which i want to avoid.

Any thoughts on how to accomplish this, or can we add a feature to line break TextFields?

**Comment by darshankawar** [comment]
@steeling 
It'd be great if you could provide `flutter doctor -v` and a [complete minimal reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) that highlights the issue you are facing.

**Comment by steeling** [comment]
This is a feature request not an issue, so not sure if I can provide an example since what I'm trying to do is not supported yet. Can you clarify which part you feel is unclear and I'd be happy to expand on that.

**Comment by darshankawar** [comment]
Thanks for clarifying @steeling. Can you check if your request is similar to [this](https://github.com/flutter/flutter/issues/30688) issue ?

**Comment by steeling** [comment]
Not quite. So imagine something like [this](https://www.thespruce.com/thmb/5SRLdvPIu7e1XhbgAHBc2k5PE1g=/735x0/d03ab930f0f65a3b58aa6770095b1fef-574e443f5f9b58516598688c.jpg)

On the second blank on the first line, if I start typing, it will push "to" to the next line. However if I type 3 words: "this is a a test", then I want only what can't fit to move to the next line, ie: "test". However `WidgetSpan` wraps the *entire* TextField in the span. It has no awareness of splitting text within the TextField, so it pushes the entire "this is a test" to the next line, which leaves a large, ugly gap on the first line.

Does that clarify?

**Comment by darshankawar** [comment]
Yes. Thanks for the clarification. Treating this as a feature request and updating title accordingly.
